subsys/net/lib/sockets/sockets.c

```diff
@@ -119,6 +119,12 @@ int _impl_zsock_socket(int family, int type, int proto)
 	}
 #endif
 
+#if defined(CONFIG_NET_SOCKETS_RAW)
+	if (family == AF_PACKET && type == SOCK_RAW) {
```

Comment:
Please dispatch only on AF_PACKET, handle type in your handler already. It would be pretty easy to support both SOCK_RAW and SOCK_DGRAM for AF_PACKET sockets, as described in http://man7.org/linux/man-pages/man7/packet.7.html

Reply:
ok.